### PR TITLE
Jenkins pipeline update - Closes #437

### LINKED ISSUE #437: Improve continuous integration pipeline
### Description

There are two issues with the current pipeline:

- Tests are failing (false negative) or the pipeline is timing out
- Build + tests procedure is taking more than a few minutes

Make the pipeline reliable and quick. 

### Motivation

- Reliable continuous integration is required to ensure quality of this project

### Acceptance Criteria

- Tests are performing stable
- Pipeline is running quickly

### Additional Information

- The Jenkins based infrastructure consists of dedicated node for the Lisk Service project.
- It might be worth to consider running the tests without Docker to reduce execution time.

### PR DESCRIPTION
**Pull request by MichalTuleja**
### What was the problem?

This PR resolves #437

### How was it solved?

- [x] Remove extra hosts
- [x] Add port-based checks
- [x] Migrate to PM2
- [x] Update Jenkins-specific files
- [x] Update docker-compose files

### How was it tested?

Jenkins

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,30 +1,74 @@
...
+					runServiceIfMissing('Redis', './jenkins/redis', REDIS_PORT)
+
+					// Install PM2
+					nvm(getNodejsVersion()) {
+						sh 'npm i -g pm2'
+					}
+
+					// Show env exports
+					echo 'Printing all exports'
+					sh 'export'
```
Comment (fchavant) [inline Jenkinsfile:65]:
```suggestion
					sh 'export -p'
```
was this meant for debugging maybe? if not your should be very careful not to expose secrets that way.

File: docker/lisk-service/docker-compose.core.yml [hunk 2]
```diff
@@ -11,11 +11,14 @@ services:
...
+      - LISK_CORE_HTTP=${LISK_CORE_HTTP} # legacy (core2)
+      - LISK_CORE_WS=${LISK_CORE_WS}
     env_file:
-      - ./lisk_service/env/common.env
-      - ./lisk_service/env/core.env
-    extra_hosts:
-      - "host.docker.internal:host-gateway"
+      - ./lisk-service/env/common.env
+      - ./lisk-service/env/core.env
+    # extra_hosts:
```
Comment (fchavant) [inline docker/lisk-service/docker-compose.core.yml:20]: leftover from development?

File: docker/lisk-service/env/mysql.env [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# MySQL install
+MYSQL_ROOT_PASSWORD=password
+MYSQL_DATABASE=lisk
+MYSQL_USER=lisk
+MYSQL_PASSWORD=password
```
Comment (fchavant) [inline docker/lisk-service/env/mysql.env:5]: Missing new line at EOF.

File: docker/network/core2-mainnet.env [hunk 4]
```diff
@@ -3,3 +3,7 @@
 # Lisk ecosystem configuration
 LISK_CORE_HTTP=https://mainnet.lisk.io
 LISK_CORE_WS=wss://mainnet.lisk.io
+
+# Enabled Gateway APIs
+ENABLE_HTTP_API=http-version1,http-version1-compat,http-status
+ENABLE_WS_API=rpc,rpc-v1,blockchain
```
Comment (fchavant) [inline docker/network/core2-mainnet.env:9]: Missing new line at EOF.

File: docker/network/core2-testnet.env [hunk 5]
```diff
@@ -3,3 +3,7 @@
 # Lisk ecosystem configuration
 LISK_CORE_HTTP=https://testnet.lisk.io
 LISK_CORE_WS=wss://testnet.lisk.io
+
+# Enabled Gateway APIs
+ENABLE_HTTP_API=http-version1,http-version1-compat,http-status
+ENABLE_WS_API=rpc,rpc-v1,blockchain
```
Comment (fchavant) [inline docker/network/core2-testnet.env:9]: Missing new line at EOF.

File: docker/network/core3-cd.env [hunk 6]
```diff
@@ -0,0 +1,4 @@
+## External services
+
+# Placeholder file
+# Update configuration in the Jenkinsfile.deployment
```
Comment (fchavant) [inline docker/network/core3-cd.env:4]: Missing new line at EOF.

File: docker/network/devnet.env [hunk 7]
```diff
@@ -0,0 +1,8 @@
+## External services
+
+# Lisk ecosystem configuration
+LISK_CORE_WS=ws://host.docker.internal:4001
+
+# Enabled Gateway APIs
+ENABLE_HTTP_API=http-status,http-version2
+ENABLE_WS_API=blockchain,rpc-v2
```
Comment (fchavant) [inline docker/network/devnet.env:8]: Missing new line at EOF.

File: jenkins/lisk-core/Makefile [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+down:
+	$(compose) down --volumes --remove-orphans
+
+logs:
+	$(compose) logs
+
+clean: down
+
+cli-%:
+	$(compose) exec $* /bin/sh
```
Comment (fchavant) [inline jenkins/lisk-core/Makefile:18]: Missing new line at EOF.

File: jenkins/mysql/docker-compose.yml [hunk 9]
```diff
@@ -15,8 +15,6 @@ services:
       MYSQL_DATABASE: ${ENV_LISK_DB_DATABASE}
       MYSQL_USER: ${ENV_LISK_DB_USER}
       MYSQL_PASSWORD: ${ENV_LISK_DB_PASSWORD}
-    expose:
-      - 3306
     ports:
       - "3306:3306"
```
Comment (fchavant) [inline jenkins/mysql/docker-compose.yml:19]: Is there a reason not to bind to localhost only?

```suggestion
      - "127.0.0.1:3306:3306"
```

**Review by fchavant — CHANGES_REQUESTED** [review]
Good job on making CI faster.

**Review by fchavant — APPROVED** [review]

**Review by nagdahimanshu — APPROVED** [review]
LGTM!

**Review by sameersubudhi — APPROVED** [review]

**Pull request merged**